apps/spruce/src/components/Pagination/index.tsx

```diff
@@ -28,10 +28,8 @@ const Pagination: React.FC<Props> = ({
   pageSize,
   totalResults,
 }) => {
-  const updateQueryParams = useUpdateURLQueryParams();
-  const handleChange =
-    onChange ||
-    ((page: number) => updateQueryParams({ page: page.toString() }));
+  const { setPage } = useTablePagination();
+  const handleChange = onChange || ((p: number) => setPage(p));
```

Comment:
[nit] I think you can do this
```suggestion
  const handleChange = onChange || setPage;
```